src/main/java/seedu/address/model/module/Deadline.java

```diff
@@ -16,19 +19,51 @@ public class Deadline {
     public Deadline(String deadline) {
         //deadline is optional and hence we do not requireNonNull.
         //requireNonNull(deadline);
-        value = deadline;
+        value = convertStringToDate(deadline);
     }
 
     /**
      * Returns if a given string is a valid timeSlot.
      */
     public static boolean isValidDeadline(String test) {
+        if (test.equals("None.")) {
+            return true;
+        }
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns the conversion of String to a LocalDateTime
+     * @return LocalDateTime instance
+     */
+    private LocalDateTime convertStringToDate(String deadline) {
+        if (deadline.equals("None.")) {
+            return null;
+        }
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("ddMMyy HH:mm");
+        return LocalDateTime.parse(deadline, dateTimeFormatter);
+    }
+
+    /**
+     * Returns String of desired display format
+     * @return Display format String
+     */
+    public String displayFormat() {
```

Comment:
good way to differentiate between the displayed formatting and the original value inserted